### PR TITLE
tower frontend:unprivileged env vars docs

### PR DESCRIPTION
**Pull request by bebosudo**
Add notes about the new `frontend:${TAG}-unprivileged` image created by https://github.com/seqeralabs/nf-tower-cloud/pull/4735.

I revisited the notes about adding SSL support too (visit https://help.tower.nf/23.1/enterprise/configuration/ssl_tls/#tls-version-support > "Configure Tower to present a SSL/TLS certificate" > "Modify `frontend` container") and removed some outdated bits.

@llewellyn-sl when searching for "ssl" in the search bar of https://help.tower.nf I also found this docs page: https://help.tower.nf/23.1/enterprise/_templates/nginx/cert_on_frontend/
Which seems to be a clone of the text in the "Modify `frontend` container" section above. Is this expected? Is mkdocs "building" these pages to later include them in the section above as snippets?
The problem is that I found the duplicated source text in the git repo twice:
- https://github.com/seqeralabs/nf-tower-docs/blob/bce3e4a43a62c9c64f042636ba4163cac2413031/docs/enterprise/configuration/ssl_tls.md > "Configure Tower to present a SSL/TLS certificate" > "Modify `frontend` container"
- https://github.com/seqeralabs/nf-tower-docs/blob/bce3e4a43a62c9c64f042636ba4163cac2413031/docs/enterprise/_templates/nginx/cert_on_frontend.md
So should I manually keep these two copies in sync? I only edited the text under `/docs/enterprise/configuration/ssl_tls.md`

Let me know!

### CODE REVIEW

File: docs/enterprise/configuration/ssl_tls.md [hunk 1]
```diff
@@ -184,77 +184,45 @@ You can secure your Tower implementation with a TLS certificate in several ways.
 
           2. Generate a [private certificate and key](https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs).
 
-          3. Create a `ssl.conf` file.
-
-              ```conf
+          3. Make a local copy of the `/etc/nginx/templates/tower.conf.template` file from the `frontend` container, or create a ConfigMap to store it, if you're using Kubernetes.
```
Comment (llewellyn-sl) [inline docs/enterprise/configuration/ssl_tls.md:187]:
```suggestion
          3. Make a local copy of the `/etc/nginx/templates/tower.conf.template` file from the `frontend` container, or create a ConfigMap to store it if you're using Kubernetes.
```

File: docs/enterprise/configuration/ssl_tls.md [hunk 2]
```diff
@@ -184,77 +184,45 @@ You can secure your Tower implementation with a TLS certificate in several ways.
...
-              include /etc/nginx/ssl.conf;
+                  listen ${NGINX_LISTEN_PORT} ssl default_server;
+                  listen [::]:${NGINX_LISTEN_PORT_IPV6} ssl default_server;
+
+                  ssl_certificate /etc/ssl/testcrt.crt;
+                  ssl_certificate_key /etc/ssl/testkey.key;
               ```
 
-          6. Modify the `frontend` container definition in your `docker-compose.yml` file:
+          5. Modify the `frontend` container definition in your `docker-compose.yml` file, or similarly for a Kubernetes manifest:
```
Comment (llewellyn-sl) [inline docs/enterprise/configuration/ssl_tls.md:199]:
```suggestion
          5. Modify the `frontend` container definition in your `docker-compose.yml` file or Kubernetes manifest:
```

File: docs/enterprise/kubernetes.md [hunk 3]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
 kubectl apply -f tower-svc.yml
 ```
 
+Note that an unprivileged version of the Tower frontend image is also available, which listens on
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:165]:
```suggestion
Note that an unprivileged version of the Tower frontend image is also available. This image listens on
```

File: docs/enterprise/kubernetes.md [hunk 4]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+an unprivileged port and therefore doesn't need to run as the root user.
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:166]:
```suggestion
an unprivileged port and therefore doesn't need to be run as the root user.
```

File: docs/enterprise/kubernetes.md [hunk 5]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+Replace the tag of the frontend image `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.1.0`
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:167]:
```suggestion
Replace the tag of the frontend image `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.x.x`
```

File: docs/enterprise/kubernetes.md [hunk 6]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+with `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.1.0-unprivileged`, and in the
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:168]:
```suggestion
with `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.x.x-unprivileged`. In the
```

File: docs/enterprise/kubernetes.md [hunk 7]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+`frontend` service below specify the `targetPort` to match the one defined with the environment
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:169]:
```suggestion
`frontend` service below, specify the `targetPort` to match the environment
```

File: docs/enterprise/kubernetes.md [hunk 8]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+variable `NGINX_LISTEN_PORT` (see below), like so:
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:170]:
```suggestion
variable `NGINX_LISTEN_PORT` (see below):
```

File: docs/enterprise/kubernetes.md [hunk 9]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+  type: LoadBalancer
+  ports:
+    - port: 80
+      containerPort: 8081
+  selector:
+    app: "frontend"
+```
+
+The new unprivileged image will soon deprecate the current image that requires root, and can be
+easily customized, e.g. some options can be defined using environment variables:
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:202]:
```suggestion
easily customized, e.g., some options can be defined using environment variables:
```

File: docs/enterprise/kubernetes.md [hunk 10]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+- `NGINX_LISTEN_PORT` (default `8000`): to specify the port the nginx process will listen on
+  inside the container; the `containerPort` on the `frontend` service must match with the value
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:204]:
```suggestion
  inside the container; the `containerPort` on the `frontend` service must match the value
```

File: docs/enterprise/kubernetes.md [hunk 11]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+  defined for the environment variable.
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:205]:
```suggestion
  defined in the environment variable.
```

File: docs/enterprise/kubernetes.md [hunk 12]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+- `NGINX_LISTEN_PORT_IPV6` (default `8000`): same as `NGINX_LISTEN_PORT`, but for the port to open
+  on the IPv6 address.
+- `NGINX_UPSTREAM_HOST` (default `backend`): the hostname of the backend service that the nginx
+  process will route requests to.
+- `NGINX_UPSTREAM_PORT` (default `8080`): the port that the backend service is exposed to.
+
+If further customizations of the config file are needed, mount a ConfigMap/Secret over the
+templated nginx configuration file at `/etc/nginx/templates/tower.conf.template`: see the
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:213]:
```suggestion
templated nginx configuration file at `/etc/nginx/templates/tower.conf.template`: see
```

**Review by llewellyn-sl — CHANGES_REQUESTED** [review]
Minor edits. Looks good otherwise.

**Comment by llewellyn-sl** [comment]
@bebosudo I believe that https://github.com/seqeralabs/nf-tower-docs/blob/bce3e4a43a62c9c64f042636ba4163cac2413031/docs/enterprise/_templates/nginx/cert_on_frontend.md was used in a previous iteration of the SSL configuration page. I will remove it from the docs since it is not being referenced on any pages.

File: docs/enterprise/kubernetes.md [hunk 13]
```diff
@@ -162,6 +162,59 @@ Deploy to your cluster:
...
+kind: Deployment
+metadata:
+  name: frontend
+  labels:
+    app: frontend
+spec:
+  ...
+      containers:
+        - name: frontend
+          image: cr.seqera.io/private/nf-tower-enterprise/frontend:v23.1.0-unprivileged
```
Comment (llewellyn-sl) [inline docs/enterprise/kubernetes.md:183]:
```suggestion
          image: cr.seqera.io/private/nf-tower-enterprise/frontend:v23.x.x-unprivileged
```

**Review by llewellyn-sl — APPROVED** [review]

**Comment by pditommaso** [comment]
@llewellyn-sl merging this PR now this repo is ahead of nf-tower-cloud/docs. How are you planning to keep in sync ?

**Comment by llewellyn-sl** [comment]
@pditommaso These were in-flight PRs at the time of the merge. I will manually push these and the configuration changes to the Cloud content. All new content will now be created in the cloud repo

**Pull request merged**